src/services/updateTransferSteps.js

```diff
@@ -19,68 +19,142 @@ const hubContract = new web3.eth.Contract(
   HubContract.abi,
   process.env.HUB_ADDRESS,
 );
+const { Edge } = require('../models/edges');
 
-// All the steps are updated
-async function updateSteps(result) {
-  const edgeUpdateManager = new EdgeUpdateManager();
-
-  const values = await Promise.allSettled(
-    result.transferSteps.map(async (step) => {
-      const tokenAddress = await hubContract.methods
-        .userToToken(step.token)
-        .call();
-
-      // Update the edge
-      await edgeUpdateManager.updateEdge(
-        {
-          token: step.token,
-          from: step.from,
-          to: step.to,
-        },
-        tokenAddress,
-      );
-
-      return true;
-    }),
-  );
+const { Op } = require('sequelize');
+const day = 1000 * 60 * 60 * 24;
 
+// Update given edges
+async function updateEdges(edges) {
+  const edgeUpdateManager = new EdgeUpdateManager();
+  for await (const edge of edges) {
+    const tokenAddress = await hubContract.methods
+      .userToToken(edge.token)
+      .call();
+    await edgeUpdateManager.updateEdge(
+      {
+        ...edge,
+      },
+      tokenAddress,
+    );
+  }
   // Write edges.csv file to update edges
-  submitJob(tasks.exportEdges, 'exportEdges-updateSteps');
+  submitJob(tasks.exportEdges, 'exportEdges-updateEdges');
+}
+
+// Get edges given an address and its limit
+async function getUserEdges(safeAddress, limit) {
+  return await Edge.findAll({
+    order: [['updatedAt', 'ASC']],
+    limit: limit,
+    raw: true,
+    where: {
+      [Op.or]: [{ from: safeAddress }, { to: safeAddress }],
+    },
+  });
+}
 
-  return values.every((step) => step.status === 'fulfilled');
+async function updateSafeEdges(safeAddress, limit = 10000) {
```

Comment:
I would say it's not necessary to sort by date here. Because the set of edges we want to update is a limited set. And we want to update all of them. The iteration of edges based on the date was necessary when updating all the edges of the database.

Reply:
updated